packages/bootstrap-vue-next/src/components/BButton/BButton.vue

```diff
@@ -6,21 +6,12 @@
     v-bind="computedAttrs"
     @click="clicked"
   >
-    <div
-      v-if="loadingBoolean"
-      class="btn-loading"
-      :class="{'mode-fill': loadingMode === 'fill', 'mode-inline': loadingMode === 'inline'}"
-    >
-      <slot name="loading">
-        <BSpinner class="btn-spinner" :small="size !== 'lg'" />
-      </slot>
-    </div>
-    <div
-      class="btn-content"
-      :class="{'btn-loading-fill': loadingBoolean && loadingMode === 'fill'}"
-    >
+    <slot v-if="loadingBoolean" name="loading">
+      <BSpinner :small="size !== 'lg'" />
```

Comment:
```suggestion
      <BSpinner :small="size !== 'lg'">
        <slot name="loadingLabel">
          {{ loadingLabel }}
        </slot>
      </BSpinner>
```
Lint & add the prop for the above comment

Reply:
This actually brings a unique issue as if the loading mode is inline, then you'd have two items of text.

So the best I would think is to have the label be on the spinner, then add another span text with a vif that changes the text as well

Reply:
Unfortunately, I don't quite understand it yet. I'll take a look at it again tomorrow.

Reply:
> This actually brings a unique issue as if the loading mode is inline, then you'd have two items of text read by a screen reader
> 
> So the best I would think is to have the label be on the spinner, then add another span text with a vif that changes the text as well
> 
> So the spinner will not announce the spinner text when inline, but instead announce the loading text span. If that makes sense

To clarify the suggested change:

This component should have three states for the Button:

1. Button is not in a loading state:
   "Button, Click me"

2. Button is in a loading state, displayed inline:
   "Button, loading, Click me"

3. Button is in a loading state, displayed as a filled loading indicator:
   "Button, loading"

Or am I wrong?

Reply:
```html
    <template v-if="loadingBoolean">
      <template v-if="!loadingFillBoolean">
        {{ loadingText }}
      </template>
      <slot name="loading">
        <BSpinner :small="size !== 'lg'" :label="loadingFillBoolean ? loadingText : undefined" />
      </slot>
    </template>
    <template v-else>
      <slot />
    </template>
```

This is my complete product,

```html
<template>
  <component
    :is="computedTag"
    class="btn"
    :class="computedClasses"
    :aria-disabled="nonStandardTag ? disabledBoolean : null"
    :aria-pressed="isToggle ? pressedBoolean : null"
    :autocomplete="isToggle ? 'off' : null"
    :disabled="isButton ? disabledBoolean : null"
    :href="props.href"
    :rel="computedLink ? props.rel : null"
    :role="nonStandardTag || computedLink ? 'button' : null"
    :target="computedLink ? props.target : null"
    :type="isButton ? props.type : null"
    :to="!isButton ? props.to : null"
    :append="computedLink ? props.append : null"
    :active-class="isBLink ? activeClass : null"
    :event="isBLink ? event : null"
    :replace="isBLink ? replace : null"
    :router-component-name="isBLink ? routerComponentName : null"
    :router-tag="isBLink ? routerTag : null"
    @click="clicked"
  >
    <template v-if="loadingBoolean">
      <template v-if="!loadingFillBoolean">
        {{ loadingText }}
      </template>
      <slot name="loading">
        <BSpinner :small="size !== 'lg'" :label="loadingFillBoolean ? loadingText : undefined" />
      </slot>
    </template>
    <template v-else>
      <slot />
    </template>
  </component>
</template>

<script setup lang="ts">
import {computed} from 'vue'
import BSpinner from '../BSpinner.vue'
import {useBooleanish} from '../../composables'
import type {BLinkProps, Booleanish, ButtonType, ButtonVariant, Size} from '../../types'
import {isLink} from '../../utils'
import BLink from '../BLink/BLink.vue'
import {useVModel} from '@vueuse/core'

defineSlots<{
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  default?: (props: Record<string, never>) => any
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  loading?: (props: Record<string, never>) => any
}>()

const props = withDefaults(
  defineProps<
    {
      pill?: Booleanish
      pressed?: Booleanish
      size?: Size
      squared?: Booleanish
      tag?: string
      type?: ButtonType
      variant?: ButtonVariant | null
      loading?: Booleanish
      loadingFill?: Booleanish
      block?: Booleanish
      loadingText?: string
    } & Omit<BLinkProps, 'variant'>
  >(),
  {
    active: false,
    pill: false,
    pressed: undefined,
    size: 'md',
    squared: false,
    tag: 'button',
    type: 'button',
    variant: 'secondary',
    loading: false,
    loadingFill: false,
    block: false,
    loadingText: 'Loading...',
    // Link props
    activeClass: 'router-link-active',
    append: false,
    disabled: false,
    event: 'click',
    href: undefined,
    // noPrefetch: {type: [Boolean, String] as PropType<Booleanish>, default: false},
    // prefetch: {type: [Boolean, String] as PropType<Booleanish>, default: null},
    rel: undefined,
    replace: false,
    routerComponentName: 'router-link',
    routerTag: 'a',
    target: '_self',
    to: undefined,
    opacity: undefined,
    opacityHover: undefined,
    underlineVariant: null,
    underlineOffset: undefined,
    underlineOffsetHover: undefined,
    underlineOpacity: undefined,
    underlineOpacityHover: undefined,
    icon: false,
    // End link props
  }
)

const emit = defineEmits<{
  'click': [value: MouseEvent]
  'update:pressed': [value: boolean]
}>()

const pressedValue = useVModel(props, 'pressed', emit)

const activeBoolean = useBooleanish(() => props.active)
const blockBoolean = useBooleanish(() => props.block)
const disabledBoolean = useBooleanish(() => props.disabled)
const pillBoolean = useBooleanish(() => props.pill)
const pressedBoolean = useBooleanish(() => props.pressed)
const squaredBoolean = useBooleanish(() => props.squared)
const loadingBoolean = useBooleanish(() => props.loading)
const loadingFillBoolean = useBooleanish(() => props.loadingFill)

const isToggle = computed<boolean>(() => typeof pressedBoolean.value === 'boolean')
const isButton = computed<boolean>(
  () => props.tag === 'button' && props.href === undefined && props.to === undefined
)
const computedLink = computed<boolean>(() => isLink(props))
const isBLink = computed<boolean>(() => props.to !== undefined)
const nonStandardTag = computed<boolean>(() => (props.href !== undefined ? false : !isButton.value))

const computedClasses = computed(() => [
  [`btn-${props.size}`],
  {
    [`btn-${props.variant}`]: props.variant !== null,
    'btn-block': blockBoolean.value,
    'active': activeBoolean.value || pressedBoolean.value,
    'rounded-pill': pillBoolean.value,
    'rounded-0': squaredBoolean.value,
    'disabled': disabledBoolean.value,
  },
])

const computedTag = computed<string | typeof BLink>(() =>
  isBLink.value ? BLink : props.href ? 'a' : props.tag
)

const clicked = (e: MouseEvent): void => {
  if (disabledBoolean.value) {
    e.preventDefault()
    e.stopPropagation()
    return
  }
  emit('click', e)
  if (isToggle.value) {
    pressedValue.value = !pressedBoolean.value
  }
}
</script>
```

(Mind that I made some merge conflicts)

Feel free to copy/paste this code into your PR so we can push this forward.

Reply:
It contains an issue where when you go to loading, it will alter the size of the button, but this is a different issue that can be fixed later... If it can be considered a "bug"

Reply:
Just pushed your Changes. Sorry that I could not follow